### PR TITLE
Add withDebuggerDebounce

### PR DESCRIPTION
**Pull request by alfonsogarciacaro**
This PR adds a new `withDebuggerEvery` helper. Its purpose is to restrict connection to the debugger to every X state updates. This is because games with physics, for example, can have many frequent states updates (up to 60 times per second) and this generates a lot of noise in the debugger, so it's nice to have an option to connect only 2 or 3 times per second.

The name of the helper is arguable, please feel free to change it if you want ;)

### CODE REVIEW

**Comment by alfonsogarciacaro** [comment]
BTW, it seems the file `Meta.props` is not committed to the repo. It gave me an error when building the project.

**Comment by MangelMaxime** [comment]
@alfonsogarciacaro The `Meta.props` should be generated by fake as we (at least me) always use Fake to test the libs build etc.

Not sure of @et1975 position.

**Comment by et1975** [comment]
Correct, all Elmish projects generate meta.props using RELEASE_NOTES.

**Review by et1975 — APPROVED** [review]
:+1:

**Comment by alfonsogarciacaro** [comment]
Sorry, please don't merge this yet. I'm thinking now that this solution maybe _too simple_. Perhaps it's better to use something like [ReactiveX throttle/debounce](http://reactivex.io/documentation/operators/debounce.html).

**Comment by alfonsogarciacaro** [comment]
I replaced the new helper with `withDebuggerDebounce`. This is probably more useful so users can control how many state updates are saved per second (with the previous method that was rather arbitrary). The behaviour is slightly different from ReactiveX `debounce`, but nobody ever understood how ReactiveX worked so it should be fine :wink:

**Comment by alfonsogarciacaro** [comment]
Quick question: why the model has to be deflated before calling `connection.init` but not `connection.send`?

**Comment by michaelsg** [comment]
I found that issue with connection.init vs connection.send. The state has to be encoded the same way in both places.

I have a fixed debugger, but it contains a few more changes than that one.
1) I have gone to using toJsonWithTypeInfo/ofJsonWithTypeInfo.
   While none of the examples show this, were the model to contain any derived instances or interface implementations, restoring the model w/o type info will be impossible.
2) I have coded a work-around for using the Chrome App as the display program.
   I have a question into RemoteDev now as to why this behaves differently to the program hosted at remotedev.io/local.
3) Better encoding of Action info - flattens the display hierarchy in debugger UI.

**Comment by et1975** [comment]
I'm thinking we need to make the serialization parametrizable, as I don't see myself ever wanting to serialize interfaces and not willing to include the cost for it into the base lib.

**Comment by michaelsg** [comment]
Not just interface implementations but any derived instance.
type A = class end
type B = inherit A

model contains Array of A  with some instances being A and some being B.
Deserializing w/o type info will assume all instances in the array are of type A (wrong).

**Comment by et1975** [comment]
Sorry, I don't mean to seem contrarian, but I don't see myself using inheritance either. I'm not saying no one should use it either, so I'm open to making the serialization pluggable. However, imho the state should be just data, and modelling the data with inheritance is unnecessary and is simply asking for this kind of trouble.

**Comment by alfonsogarciacaro** [comment]
@michaelsg proposals look very interesting, though I agree in the case of serialization it may be better to make it parametrizable. In any case, that probably deserves its own PR, should we accept this one?

**Review by et1975 — APPROVED** [review]
I'm thinking the next option for debugger will require redesign of how they are supplied and overriden. Something like how the `Program` itself is parametrized should be possible for the debugger options.

**Comment by alfonsogarciacaro** [comment]
Should I interpret this as an OK to push the MERGE button? :wink: It looks that I can.

**Comment by et1975** [comment]
Sorry, yes, absolutely, you have the powers!

**Comment by et1975** [comment]
published as beta-2

**Pull request merged**
